crates/bins/src/bin/datadog_static_analyzer_server/ide/configuration_file/endpoints.rs

```diff
@@ -0,0 +1,63 @@
+use super::error::ConfigFileError;
+use super::models::{AddRuleSetsRequest, IgnoreRuleRequest};
+use super::static_analysis_config_file::StaticAnalysisConfigFile;
+use kernel::utils::encode_base64_string;
+use rocket::http::Status;
+use rocket::serde::json::Json;
+use tracing::instrument;
+
+#[instrument()]
+#[rocket::post(
+    "/v1/config/ignore-rule",
+    format = "application/json",
+    data = "<request>"
+)]
+pub fn ignore_rule(
+    request: Json<IgnoreRuleRequest>,
+) -> Result<Json<String>, rocket::response::status::Custom<ConfigFileError>> {
+    let IgnoreRuleRequest {
+        rule,
+        configuration_base64,
+        encoded,
+        ..
+    } = request.into_inner();
+    tracing::debug!(rule, content = &configuration_base64);
+    let result = StaticAnalysisConfigFile::with_ignored_rule(rule.into(), configuration_base64);
+    to_json_result(result, encoded)
+}
+
+#[instrument()]
+#[rocket::post("/v1/config/rulesets", format = "application/json", data = "<request>")]
+pub fn post_rulesets(
+    request: Json<AddRuleSetsRequest>,
+) -> Result<Json<String>, rocket::response::status::Custom<ConfigFileError>> {
+    let AddRuleSetsRequest {
+        rulesets,
+        configuration_base64,
+        encoded,
+        ..
+    } = request.into_inner();
+    tracing::debug!(
+        rulesets=?&rulesets,
+        content=&configuration_base64
+    );
+    let result = StaticAnalysisConfigFile::with_added_rulesets(&rulesets, configuration_base64);
+    to_json_result(result, encoded)
+}
+
+#[instrument()]
+#[rocket::get("/v1/config/rulesets/<content>", format = "application/json")]
+pub fn get_rulesets(content: &str) -> Json<Vec<String>> {
```

Comment:
If the file is incorrectly formatted, this endpoint should be able to return an error (with 500 status code), so we can tell the user that SA is disabled because the config file is invalid. Otherwise, they don't have any way to know.

An alternative would be to add a separate `/config/validate` endpoint. No strong preference.

Reply:
Yes, we could extend it like that. I didn't go as far as that for this PR because I thought that if the endpoint returned an empty collection that would implicitly mean that the parsing went wrong. The IDE knows that the file is there, hence the only option is that either the file is empty or is bad formatted. In both cases, it could be seen as an invalid config file.

Reply:
I still find it a bit cleaner to return an error status code, although your solution works. It however limits us in a future if we want to eg make this function only return the rulesets that apply to a file, after filtering the ignores/only. But no strong opinion, we could also bump the api version later if we implement that behavior, so up to you.

Reply:
I think I'll leave it as it is right now and evolve it as needed. The case you're picturing, I guess we would have to change the signature anyways because we would need to pass the file path or something. So that would make us bump either the version, or think of a different endpoint. Is that right? If not, it's not it's not too much work to change it.